src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -243,80 +337,36 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
-    return CompletableFuture.allOf(requestQueueFutures.toArray(new CompletableFuture[requestQueueFutures.size()]))
-      .thenApply(x -> {
-        Map<Item, RequestQueue> itemQueueSizeMap = new HashMap<>();
-        List<Item> failedQueuesItemList = new LinkedList<>(); //to preserve the order it was found.
-
-        for (Map.Entry<Item, CompletableFuture<Result<RequestQueue>>> entry : itemRequestQueueMap.entrySet()) {
-          Result<RequestQueue> requestQueueResult = entry.getValue().join();
-          if (requestQueueResult.succeeded()) {
-            itemQueueSizeMap.put(entry.getKey(), requestQueueResult.value());
-          } else {
-            failedQueuesItemList.add(entry.getKey());
-          }
+        if (itemDueDateMap == null && itemQueueSizeMap == null) {
+          return records;
         }
 
-        if (failedQueuesItemList.size() == requestQueueFutures.size()
-            && (records.getSortedAvailableItems() == null || records.getSortedAvailableItems().isEmpty())) {
-          //fail the requests when there are no items to make requests from.
-          log.error("Failed to find request queues for all items of instanceId {}",
-                        unsortedUnavailableItems.get(0).getInstanceId());
-          return failed(new ServerErrorFailure("Unable to find an item to place a request"));
-        }
         //Sort the map
-        Map<Item, RequestQueue> sortedMap = sortRequestQueues(itemQueueSizeMap);
+        Map<Item, Integer> sortedMap = sortRequestQueues(itemQueueSizeMap, itemDueDateMap,
+          records.getInstanceLevelRequest().getPickupServicePointId());
 
-        LinkedList<Item> finalOrdedList = new LinkedList<>();
-
-        finalOrdedList.addAll(sortedMap.keySet());
-        //put the items that weren't able to retrieve RequestQueues for on the bottom of the list
-        finalOrdedList.addAll(failedQueuesItemList);
+        LinkedList<Item> finalOrdedList = new LinkedList<>(sortedMap.keySet());
 
+        //put the items that it wasn't able to retrieve loans or RequestQueues for on the bottom of the list
+        finalOrdedList.addAll(records.getItemsWithoutLoans());
+        finalOrdedList.addAll(records.getItemsWithoutRequests());
         records.setSortedUnavailableItems(finalOrdedList);
-
-        return succeeded(records);
-      });
-  }
-
-  private static Map<Item, RequestQueue> sortRequestQueues(Map<Item, RequestQueue> unsortedItems) {
-    return unsortedItems
-      .entrySet()
-      .stream()
-      .sorted(compareQueueLengths())
-      .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue,
-        (oldValue, newValue) -> oldValue, (LinkedHashMap::new)));
-  }
-
-  private static Comparator<Map.Entry<Item, RequestQueue>> compareQueueLengths() {
-    // Sort the list
-    return (q1, q2) -> {
-      RequestQueue queue1 = q1.getValue();
-      RequestQueue queue2 = q2.getValue();
-
-      return queue1.size() - queue2.size();
-    };
+        return records;
```

Comment:
Why not remove this `return records;` and let the `return records;` in the outer block handle the return?

Reply:
Done